### PR TITLE
Adapt to  CUDA.jl v3

### PR DESCRIPTION
**Pull request by nirmal-suthar**

### CODE REVIEW

File: src/rep/mesh.jl [hunk 1]
```diff
@@ -131,8 +131,8 @@ function TriMesh(
     verts = [T.(v) for v in verts]
     faces = [R.(f) for f in faces]
 
-    _verts_len = S(size.(verts, 2))
-    _faces_len = S(size.(faces, 2))
+    _verts_len = cpu(size.(verts, 2))
```
Comment (DhairyaLGandhi) [inline src/rep/mesh.jl:134]: Why do you need to call `cpu` here? Shouldn't the broadcast produce a cpu vector always?

File: test/cuda/metrics.jl [hunk 2]
```diff
@@ -66,7 +66,7 @@
             end
         end
 
-        verts = get_verts_packed(m)
+        verts = get_verts_packed(m) |> cpu
```
Comment (DhairyaLGandhi) [inline test/cuda/metrics.jl:69]: Why do we bring the to cpu?

Comment (nirmal-suthar) [reply test/cuda/metrics.jl:69]: These are tests; the latter test computation involved Array, which caused trouble

File: src/rep/mesh.jl (hunk 1 above)
Comment (nirmal-suthar) [reply src/rep/mesh.jl]: my bad. `_verts_len` is always on cpu.

**Comment by nirmal-suthar** [comment]
@DhairyaLGandhi Is there any badge link for gpu ci?

**Comment by DhairyaLGandhi** [comment]
`[Build status](https://buildkite.com/julialang/flux3d-dot-jl)`

[Build status](https://buildkite.com/julialang/flux3d-dot-jl)

**Pull request merged**